lib/checkother.cpp

```diff
@@ -1135,6 +1135,32 @@ void CheckOther::checkVariableScope()
     }
 }
 
+// used to check if an argument to a function might depend on another argument
+static bool mayDependOn(const ValueType *other, const ValueType *original)
+{
+    if (!other || !original)
+        return false;
+
+    // other must be pointer
+    if (!other->pointer)
+        return false;
+
+    // must be same underlying type
+    if (other->type != original->type)
+        return false;
+
+    const int otherPtr = other->pointer + (other->reference == Reference::LValue ? 1 : 0);
+    const int originalPtr = original->pointer + (original->reference == Reference::LValue ? 1 : 0);
```

Comment:
is it interesting if original pointer is a reference?

Reply:
I imagined this case (declared in analysed source file but defined in a different TU):
```
void f(const char *&dst, const char *src)
{
    dst = src;
}
```
So I'm combining them and count the total level of indirection.

Reply:
yes.. but I wondered about the "src" pointer:

```
void f(const char *&dst, const char *&src)
{
    dst = src;
}
```
I believe we should return true here also.

Reply:
Yes, I added a test for this as well.